src/python/SConscript

```diff
@@ -178,10 +178,34 @@ PySource('gem5.components.cachehierarchies.ruby.caches.mi_example',
     'dma_controller.py')
 PySource('gem5.components.cachehierarchies.ruby.caches.mi_example',
     'gem5/components/cachehierarchies/ruby/caches/mi_example/l1_cache.py')
+PySource('gem5.components.cachehierarchies.ruby.caches.viper',
+    'gem5/components/cachehierarchies/ruby/caches/viper/__init__.py')
+PySource('gem5.components.cachehierarchies.ruby.caches.viper',
+    'gem5/components/cachehierarchies/ruby/caches/viper/corepair_cache.py')
+PySource('gem5.components.cachehierarchies.ruby.caches.viper',
+    'gem5/components/cachehierarchies/ruby/caches/viper/directory.py')
+PySource('gem5.components.cachehierarchies.ruby.caches.viper',
+    'gem5/components/cachehierarchies/ruby/caches/viper/dma_controller.py')
+PySource('gem5.components.cachehierarchies.ruby.caches.viper',
+    'gem5/components/cachehierarchies/ruby/caches/viper/tcp.py')
+PySource('gem5.components.cachehierarchies.ruby.caches.viper',
+    'gem5/components/cachehierarchies/ruby/caches/viper/sqc.py')
+PySource('gem5.components.cachehierarchies.ruby.caches.viper',
+    'gem5/components/cachehierarchies/ruby/caches/viper/tcc.py')
 PySource('gem5.components.cachehierarchies.ruby.topologies',
     'gem5/components/cachehierarchies/ruby/topologies/__init__.py')
 PySource('gem5.components.cachehierarchies.ruby.topologies',
     'gem5/components/cachehierarchies/ruby/topologies/simple_pt2pt.py')
+
+PySource('gem5.components.devices',
```

Comment:
I agree that other devices may be attached some day. However, I think we shouldn't be shy of specializing things for a specific type of device (e.g., GPU).

On the CXL front, at least with the way we're doing it, it won't be attached as a device in this way. We're taking the approach of creating a new kind of board (`<ISA>CXLHostBoard`) which takes as a parameter an optional remote memory device. There's just too many special things we need to do on that board to use the same board as what the GPU is using.

In general, since there will be a lot of "device"-specific code, I think it's OK if we continue to specialize and create different boards rather than try to make "one board to rule them all."

I don't have any strong opinion on the layout here. It's up to Matt and Bobby. However, I did want to give a bit of context on what else we're working on.